File: analyze-acss.js
const fs = require('fs');

// Let's try to analyze the structure step by step
const jsonFile = '/Users/nerveband/wavedepth Dropbox/Ashraf Ali/Mac (2)/Documents/GitHub/tayyab-podcast-landing/acss-old.json';

try {
    console.log("Reading file...");
    const rawData = fs.readFileSync(jsonFile, 'utf8');
    console.log(`File size: ${rawData.length} characters`);
    
    console.log("Parsing JSON...");
    const data = JSON.parse(rawData);
    
    console.log("=== ACSS Configuration Analysis ===\n");
    
    console.log("MAIN CATEGORIES:");
    const mainKeys = Object.keys(data).sort();
    mainKeys.forEach((key, index) => {
        console.log(`${index + 1:2d}. ${key}`);
    });
    
    console.log(`\nTotal main categories: ${mainKeys.length}\n`);
    
    // Analyze each category
    console.log("=== CATEGORY DETAILS ===\n");
    
    mainKeys.forEach(category => {
        const content = data[category];
        console.log(`📁 ${category}`);
        console.log(`   Type: ${Array.isArray(content) ? 'Array' : typeof content}`);
        
        if (typeof content === 'object' && content !== null) {
            if (Array.isArray(content)) {
                console.log(`   Length: ${content.length}`);
                if (content.length > 0) {
                    console.log(`   First item: ${typeof content[0]} - ${JSON.stringify(content[0]).substring(0, 100)}...`);
                }
            } else {
                const keys = Object.keys(content);
                console.log(`   Properties: ${keys.length}`);
                if (keys.length > 0) {
                    console.log(`   Sample keys: [${keys.slice(0, 5).join(', ')}]${keys.length > 5 ? ` ...and ${keys.length - 5} more` : ''}`);
                }
            }
        } else {
            console.log(`   Value: ${String(content).substring(0, 50)}${String(content).length > 50 ? '...' : ''}`);
        }
        console.log('');
    });

} catch (error) {
    console.error(`Error: ${error.message}`);
    console.error(`Stack: ${error.stack}`);
}

File: analyze-acss.py
#!/usr/bin/env python3
import json

# Direct path to the file
json_file = '/Users/nerveband/wavedepth Dropbox/Ashraf Ali/Mac (2)/Documents/GitHub/tayyab-podcast-landing/acss-old.json'

try:
    with open(json_file, 'r') as f:
        data = json.load(f)
    
    print("=== ACSS Configuration Structure Analysis ===\n")
    
    print("MAIN CONFIGURATION CATEGORIES:")
    for key in sorted(data.keys()):
        print(f"• {key}")
    
    print("\n=== DETAILED ANALYSIS ===\n")
    
    # Analyze each main category
    for category, content in data.items():
        print(f"Category: {category}")
        print(f"Type: {type(content).__name__}")
        
        if isinstance(content, dict):
            if len(content) > 0:
                print(f"Sub-categories/properties: {len(content)}")
                # Show first few keys as examples
                sample_keys = list(content.keys())[:5]
                print(f"Sample keys: {sample_keys}")
                if len(content) > 5:
                    print(f"... and {len(content) - 5} more")
        elif isinstance(content, list):
            print(f"Array length: {len(content)}")
            if len(content) > 0:
                print(f"First item type: {type(content[0]).__name__}")
        else:
            print(f"Value: {str(content)[:100]}...")
        
        print("-" * 50)

except FileNotFoundError:
    print(f"File not found: {json_file}")
except json.JSONDecodeError as e:
    print(f"JSON decode error: {e}")
except Exception as e:
    print(f"Error: {e}")